bin/data_frame_analyzer/Main.cc

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+//! \brief
+//! Applies a range of ML analyses on a data frame.
+//!
+//! DESCRIPTION:\n
+//! Expects to be streamed CSV or length encoded data on STDIN or a named pipe,
+//! and sends its JSON results to STDOUT or another named pipe.
+//!
+//! IMPLEMENTATION DECISIONS:\n
+//! Standalone program.
+//!
+#include <core/CJsonOutputStreamWrapper.h>
+#include <core/CLogger.h>
+#include <core/CProcessPriority.h>
+#include <core/CoreTypes.h>
+
+#include <ver/CBuildInfo.h>
+
+#include <api/CDataFrameAnalyzer.h>
+#include <api/CIoManager.h>
+#include <api/CLengthEncodedInputParser.h>
+#include <api/COutputChainer.h>
+
+#include "CCmdLineParser.h"
+
+#include <boost/bind.hpp>
+#include <boost/scoped_ptr.hpp>
+
+#include <string>
+
+#include <stdlib.h>
+
+int main(int argc, char** argv) {
+    // Read command line options
+    std::string configFile;
+    std::string logProperties;
+    std::string logPipe;
+    std::string inputFileName;
+    bool isInputFileNamedPipe(false);
+    std::string outputFileName;
+    bool isOutputFileNamedPipe(false);
+    if (ml::data_frame_analyzer::CCmdLineParser::parse(
+            argc, argv, configFile, logProperties, logPipe, inputFileName,
+            isInputFileNamedPipe, outputFileName, isOutputFileNamedPipe) == false) {
+        return EXIT_FAILURE;
+    }
+
+    // Construct the IO manager before reconfiguring the logger, as it performs
+    // std::ios actions that only work before first use
+    ml::api::CIoManager ioMgr(inputFileName, isInputFileNamedPipe,
+                              outputFileName, isOutputFileNamedPipe);
+
+    if (ml::core::CLogger::instance().reconfigure(logPipe, logProperties) == false) {
+        LOG_FATAL("Could not reconfigure logging");
```

Comment:
We prefer leading << for all log messages even where not strictly needed for consistency. There are a few discrepancies in this file.